netbox/resource_netbox_device_interface.go

```diff
@@ -0,0 +1,259 @@
+package netbox
+
+import (
+	"context"
+	"regexp"
+	"strconv"
+
+	"github.com/fbreckle/go-netbox/netbox/client"
+	"github.com/fbreckle/go-netbox/netbox/client/dcim"
+	"github.com/fbreckle/go-netbox/netbox/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceNetboxDeviceInterface() *schema.Resource {
+	validModes := []string{"access", "tagged", "tagged-all"}
+
+	return &schema.Resource{
+		CreateContext: resourceNetboxDeviceInterfaceCreate,
+		ReadContext:   resourceNetboxDeviceInterfaceRead,
+		UpdateContext: resourceNetboxDeviceInterfaceUpdate,
+		DeleteContext: resourceNetboxDeviceInterfaceDelete,
+
+		Description: `:meta:subcategory:Device:From the [official documentation](https://docs.netbox.dev/en/stable/features/device/#interface):
+
+> Interfaces in NetBox represent network interfaces used to exchange data with connected devices. On modern networks, these are most commonly Ethernet, but other types are supported as well. IP addresses and VLANs can be assigned to interfaces.`,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"device_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"mac_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^([A-Z0-9]{2}:){5}[A-Z0-9]{2}$"),
```

Comment:
There is a pre-made [IsMACAddress](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/validation#IsMACAddress) filter (yes, that should also be used in the VM interface resource :P)

Reply:
thank you, has been resolved